src/mongocxx/test/spec/uri.cpp

```diff
@@ -0,0 +1,270 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fstream>
+#include <mutex>
+#include <thread>
+
+#include <bsoncxx/test_util/catch.hh>
+#include <mongocxx/exception/operation_exception.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/test/spec/monitoring.hh>
+
+namespace {
+
+static bsoncxx::document::value _doc_from_file(mongocxx::stdx::string_view sub_path) {
+    const char* test_path = std::getenv("URI_OPTIONS_TESTS_PATH");
+    REQUIRE(test_path);
+
+    std::string path = std::string(test_path) + sub_path.data();
+    CAPTURE(path);
+
+    std::ifstream file{path};
+    REQUIRE(file);
+
+    std::string file_contents((std::istreambuf_iterator<char>(file)),
+                              std::istreambuf_iterator<char>());
+
+    return bsoncxx::from_json(file_contents);
```

Comment:
The `seeds`, `numSeeds`, `parsed_options`, `numHosts` and `comment` fields in the test format [are not handled](https://github.com/mongodb/specifications/blob/0a44c5bb3133d7b3e156276a9cba45daccb9266d/source/initial-dns-seedlist-discovery/tests/README.rst#test-format-and-use).

That may be OK, since the subset of tests do not use those fields.

Suggest parsing the fields explicitly. Erroring on unhandled fields will inform us when a test is added with new fields:

```c++
struct InitialDNSSeedlistTest {
    bsoncxx::stdx::string_view uri;
    bsoncxx::array::view hosts;
    int32_t num_hosts;
    bsoncxx::document::view options;
    bool error;
    bool ping;

    static InitialDNSSeedlistTest parse(bsoncxx::document::view test_doc) {
        InitialDNSSeedlistTest test;

        for (auto el : test_doc) {
            if (0 == el.key().compare("uri")) {
                test.uri = el.get_string().value;
            } else if (0 == el.key().compare("seeds") || 0 == el.key().compare("numSeeds")) {
                // The 'seeds' and 'numSeeds' assertions are explicitly skipped. The C++ driver does
                // not have access to the initial seedlist populated in the C driver.
            } else if (0 == el.key().compare("hosts")) {
                test.hosts = el.get_array().value;
            } else if (0 == el.key().compare("numHosts")) {
                test.num_hosts = el.get_int32().value;
            } else if (0 == el.key().compare("options")) {
                test.options = el.get_document().value;
            } else if (0 == el.key().compare("error")) {
                test.error = el.get_bool().value;
            } else if (0 == el.key().compare("comment")) {
                // Ignore comment.
            } else if (0 == el.key().compare("ping")) {
                test.ping = el.get_bool().value;
            } else {
                FAIL("InitialDNSSeedlistTest does not understand the field: '"
                     << el.key() << "'. Please add support.");
            }
        }
        return test;
    }
};
```